### ISSUE TITLE
[HOLD for payment 2024-06-05] [$500] Fix scrollable elements in Policy pages

### ISSUE THREAD

**Issue by luacmartins** [Expensify/App#37782]
Coming from [this comment](https://github.com/Expensify/App/pull/37704#issuecomment-1979414799), all elements but the page header should scroll on policy pages, e.g. members, categories, tags. etc

https://github.com/Expensify/App/assets/22219519/587cd31c-cacc-424b-a3c5-b3b6a3594776

cc @Expensify/design 
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~010982d541dff55b4c</li>
        <li>Upwork Job ID: 1765759901307744256</li>
        <li>Last Price Increase: 2024-03-07</li>
<li>Automatic offers: </li>
<ul>
<li>akinwale | Contributor | 0</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @MitchExpensify</details>

**Comment by MitchExpensify** [comment]
So the whole page should scroll just like it does everywhere else, got it. I'll make this external tomorrow

**Comment by dragnoir** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

Content on the header does not scroll with the list.

### What is the root cause of that problem?

The issue is that only the Categories lis is nested into a scrollable component `<SectionList>`.

We can't just wrap the whole content including the `<SectionList>` into a `ScrollView`. This will lead to many performance issue and bacause:

The are several reasons why nested  `VirtualizedList`  is an  **Anti-pattern**. Such as:

-   The  `VirtualizedList`  cannot calculate the  **correct window size**, because the  `ScrollView`  takes up the  **entire screen**  and does not constrain its content. Therefore, the  `VirtualizedList`  will try to  **_render all the items at once_**, defeating the purpose of virtualization and potentially  **causing performance issues**  or  **crashes**.
-   The  `ScrollView`  will intercept all the  **touch events**  and  **prevent the VirtualizedList from handling them properly**. This can affect features like  **pull-to-refresh**,  **infinite scrolling**, or  **swipe actions**.
-   The  `ScrollView`  will also interfere with the  **scroll position**  and  **momentum of the** `VirtualizedList`, causing a  **janky**  and  **inconsistent**  user experience.

Also we can't just add a new new prop `ListHeaderComponent` inside `BaseSelectionList` and then simply pass it to `SectionList`. The SectionList is used by lots of other componets where we have headerMessage, search... if we pass the sticky part to `ListHeaderComponent` and in the future we need to enable the search input, the UI will get messy.

### What changes do you think we should make in order to solve the problem?

1- On [WorkspaceCategoriesPage](https://github.com/Expensify/App/blob/main/src/pages/workspace/categories/WorkspaceCategoriesPage.tsx) create a new prop headerScrollable (true/false). 

2- if true, then we render what we need from the header inside the `[renderItem](https://github.com/Expensify/App/blob/537b7c081392c3721293989a9b6517f698580607/src/components/SelectionList/BaseSelectionList.tsx#L277)` function with item 0

```diff
+ if (index  ===  0) {
  return (
    <View>
+     <NewHeaderCompoenent />
    <ListItem
    item={item}
    ....
    />
    </View>
    );
}
```

The `<NewHeaderCompoenent />` will have all the content that we need it to be scrollable and at the top (like the +Add category and Settings buttons)

3- Adjust `BaseSelectionList` to hide/show elements as required if `headerScrollable` is true.

**Comment by MitchExpensify** [comment]
Some proposals here for you @aimane-chnaif 👍

**Comment by MitchExpensify** [comment]
Bumped @aimane-chnaif 1:1

**Comment by MitchExpensify** [comment]
How is the PR coming along @Krishna2323 ?

**Comment by shawnborton** [comment]
@Krishna2323 please tag me in the PR when you raise it, thanks!

**Comment by MitchExpensify** [comment]
Waiting on @Krishna2323 to fix merge conflicts

**Comment by MitchExpensify** [comment]
Reminder set to pay, @akinwale thoughts on the BZ steps?

**Comment by akinwale** [comment]
> * [x]  [@akinwale] The PR that introduced the bug has been identified. Link to the PR:
> * [x]  [@akinwale] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
> * [x]  [@akinwale] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:

This is a new feature.

> * [x]  [@akinwale] Determine if we should create a regression test for this bug.
> * [x]  [@akinwale] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

**Regression Test Steps**
_Precondition: Members, Distance rates, Categories, Tags and Taxes screens should have multiple records to the point where the records can be scrolled_

1. Launch the Expensify app
2. Select a Collect or Control workspace from the workspaces page

**_Devices with width greater than 800px_**
* Open any one of the Members, Distance rates, Categories, Tags or Taxes screens
* Verify that the options are scrollable on each page, and the options scroll as expected

**_Devices with width less than 800px_**
* Open any one of the Members, Distance rates, Categories, Tags or Taxes screens
* Verify that the options list scrolls properly on each screen
* Verify that the page description at the top and the thead scrolls while the buttons are fixed.

Do we agree 👍 or 👎?

**Comment by akinwale** [comment]
> Reminder set to pay, @akinwale thoughts on the BZ steps?

@MitchExpensify Checklist completed.

**Comment by akinwale** [comment]
@MitchExpensify bump for payment. Thanks.

**Comment by MitchExpensify** [comment]
Paid and contract ended, thanks @akinwale !

**Comment by MitchExpensify** [comment]
Oh ok, I'm actually unsure what to do in that case. Are you expecting to be paid via Upwork or NewDot just to be sure?

**Comment by MitchExpensify** [comment]
Offer sent here [Fix scrollable elements in Policy pages #37782](https://www.upwork.com/nx/wm/offer/102681299)

**Comment by MitchExpensify** [comment]
Paid and contract ended! Thanks for your patience @Krishna2323
